docs/customruns.md

```diff
@@ -0,0 +1,419 @@
+   `Cancellation` won't work for the Custom Task.
+3. A Custom Task controller can watch for this status update
+   (i.e. `CustomRun.Spec.Status == RunCancelled`) and or `CustomRun.HasTimedOut()`
+   and take any corresponding actions (i.e. a clean up e.g., cancel a cloud build,
+   stop the waiting timer, tear down the approval listener).
+4. A Custom Task author can differentiate between a `CustomRun` cancelled by an user
+   from a Run cancellation initiated by a PipelineRun by looking at the 
+   `CustomRun.Spec.StatusMessage`. If a Run cancellation has been initiated by a
+   cancellation of a PipelineRun then `CustomRun.Spec.StatusMessage` is
+   `"CustomRun cancelled as the PipelineRun it belongs to has been cancelled."`. 
+5. Once resources or timers are cleaned up, while it is **REQUIRED** to set a
+   `conditions` on the `CustomRun`'s `status` of `Succeeded/False` with an optional
+   `Reason` of `CustomRunTimedOut`.
+6. `Timeout` is specified for each `retry attempt` instead of all `retries`.
+
+### Specifying `Retries`
+
+A custom task specification can be created with `Retries` as follows:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: Run
+metadata:
+  generateName: simpleexample
+spec:
+  retries: 3 # set retries
+  params:
+    - name: searching
+      value: the purpose of my existence
+  ref:
+    apiVersion: custom.tekton.dev/v1alpha1
+    kind: Example
+    name: exampleName
+```
+
+Supporting retries is optional but recommended.
+
+#### Developer guide for custom controllers supporting `retries`
+
+1. Teton controller only depends on `ConditionSucceeded` to determine the 
+   termination status of a `CustomRun`, therefore Custom task implementors
+   MUST NOT set `ConditionSucceeded` to `False` until all retries are exhausted.
+2. Those custom tasks who do not wish to support retry, can simply ignore it.
+3. It is recommended, that custom task should update the field `RetriesStatus`
+   of a `CustomRun` on each retry performed by the custom task.
+4. Tekton controller does not validate that number of entries in `RetriesStatus`
+   is same as specified value of retries count.
+
+### Specifying `Parameters`
+
+If a custom task supports [`parameters`](tasks.md#parameters), you can use the
+`params` field in the `CustomRun` to specify their values:
+
+```yaml
+spec:
+  params:
+    - name: my-param
+      value: chicken
+```
+
+If the custom task controller knows how to interpret the parameter value, it
+will do so. It might enforce that some parameter values must be specified, or
+reject unknown parameter values.
+
+### Specifying Workspaces and Service Account
+
+A `CustomRun` object can specify workspaces and a service account name.
+These can be used to share data, or specify the identity to run the `CustomRun`.
+Not all custom tasks will support these values.
+Consult the documentation of the custom task that you are using to determine whether these values apply.
```

Comment:
why are these two grouped together? seems that they should be their own sections 🤔

Reply:
There were another `podTemplate` grouped with these two, I deleted it. You are right, we should split them~ Updated.